### PR TITLE
Add setup instructions for TensorRT-LLM

### PR DESCRIPTION
**Pull request by linden-li**

### CODE REVIEW

File: scripts/inference/README.md [hunk 1]
```diff
@@ -236,3 +258,12 @@ PYTHONPATH=/mnt/work/FasterTransformer python scripts/inference/run_mpt_with_ft.
     --ckpt_path mpt-ft-7b/1-gpu --lib_path /mnt/work/FasterTransformer/build/lib/libth_transformer.so \
     --sample_input_file prompts.txt --sample_output_file output.txt
 ```
+
+## Running MPT with TensorRT-LLM
+
+MPT-like architectures can be used with the NVIDIA's [TensorRT-LLM](https://github.com/NVIDIA/TensorRT-LLM) library for language model inference. To do so, follow the instructions in the [examples/mpt](https://github.com/NVIDIA/TensorRT-LLM/tree/v0.6.1/examples/mpt) directory for the most recent release, which will show you how to:
```
Comment (dakinggg) [inline scripts/inference/README.md:264]:
```suggestion
MPT-like architectures can be used with NVIDIA's [TensorRT-LLM](https://github.com/NVIDIA/TensorRT-LLM) library for language model inference. To do so, follow the instructions in the [examples/mpt](https://github.com/NVIDIA/TensorRT-LLM/tree/v0.6.1/examples/mpt) directory for the most recent release, which will show you how to:
```

File: scripts/inference/README.md [hunk 2]
```diff
@@ -236,3 +258,12 @@ PYTHONPATH=/mnt/work/FasterTransformer python scripts/inference/run_mpt_with_ft.
...
+
+1. Convert an MPT HuggingFace checkpoint into the FasterTransformer format.
+2. Build a TensorRT engine with the FasterTransformer weights
+
+Using this engine, you can utilize TensorRT-LLM for fast inference. If you would like to use TensorRT-LLM as an end-to-end solution for an inference service, you can utilize the built engine with an NVIDIA Triton server backend: an example server can be found in [this repository](https://github.com/triton-inference-server/tensorrtllm_backend/tree/v0.6.1) accompanying the most recent release.
```
Comment (dakinggg) [inline scripts/inference/README.md:269]: was "built engine" supposed to be "built-in engine"?

File: scripts/inference/README.md (hunk 1 above)
Comment (dakinggg) [reply scripts/inference/README.md:264]: I'd drop the verbiage here and below about "most recent release". Its a bit confusing since you're link is to an explicit release.

**Review by dakinggg — COMMENTED** [review]
LGTM, will leave approval for Daya or Megha

File: scripts/inference/README.md (hunk 2 above)
Comment (megha95) [reply scripts/inference/README.md:269]: I'd rephrase it as "built TRT engine". Also, here again we should drop "most recent release" as suggested by Daniel above.

File: scripts/inference/README.md [hunk 3]
```diff
@@ -175,6 +188,7 @@ python convert_hf_to_onnx.py --pretrained_model_name_or_path local/path/to/huggi
 Please open a Github issue if you discover any problems!
 
 ## Converting an HF MPT to FasterTransformer
```
Comment (megha95) [inline scripts/inference/README.md:190]: Do we want to keep FT conversion scripts? afaik, https://github.com/NVIDIA/FasterTransformer is not maintained anymore by Nvidia and wont be going forward.

**Review by megha95 — APPROVED** [review]

**Comment by dakinggg** [comment]
@megha95 can we merge this?

File: scripts/inference/README.md (hunk 2 above)
Comment (megha95) [reply scripts/inference/README.md:269]: @linden-li can you pls make the suggested changes here? also, update TRT LLM link to v0.7.1?
